modules/fedimint-ln/src/lib.rs

```diff
@@ -441,6 +444,114 @@ impl ServerModule for Lightning {
         LightningDecoder
     }
 
+    fn database_version(&self) -> DatabaseVersion {
+        DatabaseVersion { version: 3 }
+    }
+
+    async fn migrate_database(&self, db: &Database, db_version: u64) -> Result<(), anyhow::Error> {
+        let mut current_migration_version = db_version;
+        while current_migration_version < self.database_version().version {
+            match current_migration_version {
+                0 => {
+                    let mut read_dbtx = db.begin_transaction().await;
+                    let mut write_dbtx = db.begin_transaction().await;
+
+                    // Upgrade MigratableKeyV1 and MigratableValueV1 to MigratableKeyV2 and MigratableValueV2
```

Comment:
I would move each migration logic to separate closure/function and store them in some `BTreeMap<DatabaseVersion, Box<FnOnce(from: &DatabaseTransaction, to: &DatabaseTransaction)> -> Box<Future<Item = Result<...>>);` kind of thing, so this can become just a loop, lookup, call.

Reply:
Hmm were you thinking of instantiating this statically? I ran into issues doing that with a BTreeMap and a async function as the value. 

Anyway, I just refactored the logic here into another function, perhaps that's good enough for now. Let me know if you see a better way of doing it.

Reply:
Nevermind, got this to work :)